pkg/config/static_manager.go

```diff
@@ -19,56 +19,37 @@ package config
 
 import (
 	"errors"
-	"fmt"
 	"sync"
 
-	"github.com/optimizely/go-sdk/pkg/config/datafileprojectconfig"
 	"github.com/optimizely/go-sdk/pkg/logging"
 	"github.com/optimizely/go-sdk/pkg/notification"
-	"github.com/optimizely/go-sdk/pkg/utils"
 )
 
 // StaticProjectConfigManager maintains a static copy of the project config
 type StaticProjectConfigManager struct {
 	projectConfig    ProjectConfig
 	optimizelyConfig *OptimizelyConfig
 	configLock       sync.Mutex
-	logger           logging.OptimizelyLogProducer
 }
 
-// NewStaticProjectConfigManagerFromURL returns new instance of StaticProjectConfigManager for URL
-func NewStaticProjectConfigManagerFromURL(sdkKey string) (*StaticProjectConfigManager, error) {
-
-	requester := utils.NewHTTPRequester(logging.GetLogger(sdkKey, "HTTPRequester"))
+// NewStaticProjectConfigManager creates a new instance of the manager with the given sdk key and some options
+func NewStaticProjectConfigManager(sdkKey string, configMangerOptions ...OptionFunc) *StaticProjectConfigManager {
 
 	logger := logging.GetLogger(sdkKey, "StaticProjectConfigManager")
-
-	url := fmt.Sprintf(DatafileURLTemplate, sdkKey)
-	datafile, _, code, e := requester.Get(url)
-	if e != nil {
-		logger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
-		return nil, e
+	staticProjectConfigManager := newConfigManager(sdkKey, logger, configMangerOptions...)
```

Comment:
Does `config.WithDatafileAccessToken` work with `NewStaticProjectConfigManager`? I think it should. Also that implies we need the logic to switch default URL template in here, probably refactored into a shared helper method.

Reply:
yes that works, also, if you initialize static manager like this:
```
cm := config.NewStaticProjectConfigManager(sdkKey, config.WithDatafileAccessToken(token), config.WithDatafileURLTemplate(config.DatafileURLTemplate))
```
for private datafile then it will not work